Bottom: 3e90b33543a7ad7f7e207d2b8560406cce1239f4
Top:    f80eb80e16768d162a49b644071e1047a9610190
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 16:56:13 +0100

draw.c: Change DrawTop() to DrawOuterPPV() taking a side parameter


---

diff --git a/src/draw.c b/src/draw.c
index 7380beb..2ea2467 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static const BoxType *clip_box = NULL;
  */
 static void Redraw (bool, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
-static void DrawTop (const BoxType *);
+static void DrawOuterPPV (int side, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
 static void DrawPlainPin (PinTypePtr, bool);
@@ -312,14 +312,15 @@ static void
 PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
 {
   int save_swap = SWAP_IDENT;
+  int side = swap_ident ? SOLDER_LAYER : COMPONENT_LAYER;
 
   gui->set_draw_faded (Output.fgGC, 1);
-  SWAP_IDENT = swap_ident;
   DrawLayerGroup (side_group, drawn_area);
-  DrawTop (drawn_area);
+  DrawOuterPPV (side, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
+  SWAP_IDENT = swap_ident;
   DrawSilk (swap_ident,
 	    swap_ident ? solder_silk_layer : component_silk_layer,
 	    drawn_area);
@@ -402,9 +403,11 @@ DrawEverything (BoxTypePtr drawn_area)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
     return;
 
+  side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
+
   /* Draw pins, pads, vias below silk */
   if (gui->gui)
-    DrawTop (drawn_area);
+    DrawOuterPPV (side, drawn_area);
   else
     {
       HoleCountStruct hcs;
@@ -566,13 +569,11 @@ pad_callback (const BoxType * b, void *cl)
 }
 
 /* ---------------------------------------------------------------------------
- * draws pins pads and vias
+ * draws outer layer pins pads and vias
  */
 static void
-DrawTop (const BoxType * screen)
+DrawOuterPPV (int side, const BoxType * screen)
 {
-  int side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
-
   if (PCB->PinOn || doing_assy)
     {
       /* draw element pins */
